src/app/machines/views/MachineList/MachineListControls/MachinesFilterAccordion/MachinesFilterOptions/MachinesFilterOptions.tsx

```diff
@@ -44,6 +45,38 @@ const MachinesFilterOptions = ({
   );
   const currentFilters = FilterMachines.getCurrentFilters(searchText);
 
+  if (group === "workloads") {
+    // Workload annotations are key/value pairs, but we're given
+    // them as a string "key:value". So we need to split them.
+    const filterKeys = filterOptions.map((filter) => {
+      return filter.key.toString().split(":")[0];
+    });
+
+    // Find the count of each workload annotation key
+    const counts = filterKeys.reduce(
+      (result: { key: string; count: number }[], currentFilterKey) => {
+        const existingFilter = result.find(
+          (filter) => filter.key === currentFilterKey
+        );
+
+        if (existingFilter) {
+          existingFilter.count++;
+        } else {
+          result.push({ key: currentFilterKey, count: 1 });
+        }
+
+        return result;
+      },
+      []
+    );
+
+    // Turn array of keys and counts into valid filter options
+    filterOptions = counts.map((filterCount) => ({
+      key: filterCount.key,
+      label: `${filterCount.key} (${filterCount.count})`,
+    }));
+  }
+
```

Comment:
A common pattern for deriving data from redux store is to place such logic in a selector.
https://redux.js.org/usage/deriving-data-selectors#calculating-derived-data-with-selectors

Double check this, but it will be most likely safe to simply update the existing `machineSelectors.filterOptions` selector including the additional mapping you created here for the workload group.